solidity/contracts/hooks/axelar/AxelarHook.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IPostDispatchHook} from "../../interfaces/hooks/IPostDispatchHook.sol";
+import {StandardHookMetadata} from "../libs/StandardHookMetadata.sol";
+import {Message} from "../../libs/Message.sol";
+
+//TODO: remove temp intermal import. for testing speed purposes only
+interface IAxelarGateway {
+    function callContract(
+        string calldata destinationChain,
+        string calldata destinationContractAddress,
+        bytes calldata payload
+    ) external;
+}
+
+interface IAxelarGasService {
+    function payNativeGasForContractCall(
+        address sender,
+        string calldata destinationChain,
+        string calldata destinationAddress,
+        bytes calldata payload,
+        address refundAddress
+    ) external payable;
+}
+
+contract AxelarHook is IPostDispatchHook {
+    using StandardHookMetadata for bytes;
+    using Message for bytes;
+
+    IAxelarGasService public immutable AXELAR_GAS_SERVICE;
+    IAxelarGateway public immutable AXELAR_GATEWAY;
+    string public DESTINATION_CHAIN;
+    string public DESTINATION_CONTRACT;
+    bytes GMP_CALL_CODE;
+
+    constructor(
+        string memory destinationChain,
+        string memory destionationContract,
+        address axelarGateway,
+        address axelarGasReceiver,
+        bytes memory gmp_call_code
+    ) {
+        DESTINATION_CHAIN = destinationChain;
+        DESTINATION_CONTRACT = destionationContract;
+        AXELAR_GATEWAY = IAxelarGateway(axelarGateway);
+        AXELAR_GAS_SERVICE = IAxelarGasService(axelarGasReceiver);
+        GMP_CALL_CODE = gmp_call_code;
+    }
+
+    /**
+     * @notice Returns an enum that represents the type of hook
+     */
+    function hookType() external pure returns (uint8) {
+        return uint8(IPostDispatchHook.Types.AXELAR);
+    }
+
+    /**
+     * @notice Returns whether the hook supports metadata
+     * @return true the hook supports metadata
+     */
+    function supportsMetadata(bytes calldata) external pure returns (bool) {
+        return true;
+    }
```

Comment:
could make this support both standard by "guessing" the axelar amount and your special variant with custom metadata

Reply:
preferably we should receive an exact amount